cloudant-client/src/main/java/com/cloudant/client/internal/query/Definition.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2017 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.client.internal.query;
+
+import com.cloudant.client.api.query.Field;
+import com.google.gson.JsonObject;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+/**
+ * Abstract model of common index definition parameters.
+ */
+public class Definition<F extends Field> {
+    //TODO decide if this alternate is needed
```

Comment:
I think we will need this to read indexes created prior to Couch 2.x so we should probably just remove this TODO.

Reply:
Done in 711ed29